integration/custom_conf_app_test.go

```diff
@@ -0,0 +1,79 @@
+package integration_test
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam"
+
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+func testCustomConfApp(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		pack   occam.Pack
+		docker occam.Docker
+
+		name      string
+		source    string
+		image     occam.Image
+		container occam.Container
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack().WithNoColor().WithVerbose()
+		docker = occam.NewDocker()
+
+		var err error
+		name, err = occam.RandomName()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	it.After(func() {
+		Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+		Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+		Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+		Expect(os.RemoveAll(source)).To(Succeed())
+	})
+
+	context("when pushing an app with a custom nginx.conf location", func() {
+		it.Before(func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "custom_conf_app"))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it("serves up staticfile", func() {
+			var err error
+			image, _, err = pack.Build.
+				WithBuildpacks(nginxBuildpack).
+				WithPullPolicy("never").
+				WithEnv(map[string]string{
+					"BP_NGINX_CONF_LOCATION": "conf/nginx.conf",
+				}).
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithEnv(map[string]string{"PORT": "8080"}).
+				WithPublish("8080").
+				Execute(image.ID)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(container).Should(BeAvailable())
+
+			response, err := http.Get(fmt.Sprintf("http://localhost:%s/index.html", container.HostPort("8080")))
```

Comment:
We can use the [`Serve` matcher in occam](https://github.com/paketo-buildpacks/occam/blob/85d6901da66daf207a7ceef437b4336f47b3041f/matchers/serve.go#L63) to make this cleaner and easier to read.

However, we should do this consistently across all tests, so I'd be happy to push that to a separate issue.

Reply:
@robdimsdale Let's make this change later.